aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/EventRecorder.kt

```diff
@@ -15,29 +15,261 @@
 package com.amplifyframework.analytics.pinpoint
 
 import android.content.Context
+import android.database.Cursor
 import android.net.Uri
 import aws.sdk.kotlin.services.pinpoint.PinpointClient
+import aws.sdk.kotlin.services.pinpoint.model.EndpointDemographic
+import aws.sdk.kotlin.services.pinpoint.model.EndpointItemResponse
+import aws.sdk.kotlin.services.pinpoint.model.EndpointLocation
+import aws.sdk.kotlin.services.pinpoint.model.EndpointUser
+import aws.sdk.kotlin.services.pinpoint.model.Event
+import aws.sdk.kotlin.services.pinpoint.model.EventItemResponse
+import aws.sdk.kotlin.services.pinpoint.model.EventsBatch
+import aws.sdk.kotlin.services.pinpoint.model.EventsRequest
+import aws.sdk.kotlin.services.pinpoint.model.PublicEndpoint
+import aws.sdk.kotlin.services.pinpoint.model.PutEventsRequest
+import aws.sdk.kotlin.services.pinpoint.model.Session
 import com.amplifyframework.analytics.AnalyticsEvent
+import com.amplifyframework.analytics.pinpoint.database.EventTable
 import com.amplifyframework.analytics.pinpoint.database.PinpointDatabase
+import com.amplifyframework.analytics.pinpoint.internal.core.util.millisToIsoDate
 import com.amplifyframework.analytics.pinpoint.models.PinpointEvent
+import com.amplifyframework.analytics.pinpoint.targeting.TargetingClient
+import com.amplifyframework.analytics.pinpoint.targeting.endpointProfile.EndpointProfile
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.logging.Logger
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 
 internal class EventRecorder(
     val context: Context,
-    val pinpointClient: PinpointClient,
+    private val pinpointClient: PinpointClient,
     private val pinpointDatabase: PinpointDatabase,
-    private val coroutineDispatcher: CoroutineDispatcher = Dispatchers.Default
+    private val targetingClient: TargetingClient,
+    private val coroutineDispatcher: CoroutineDispatcher = Dispatchers.Default,
+    private val logger: Logger =
+        Amplify.Logging.forNamespace(
+            AWS_PINPOINT_ANALYTICS_LOG_NAMESPACE.format(EventRecorder::class.java.simpleName)
```

Comment:
looks more specific than the generalized way I've been doing it in my code -- I'll implement this as well (fast-follow PR with unit tests?)